src/Maths/math.vector.ts

```diff
@@ -3452,6 +3452,34 @@ export class Quaternion {
         return result;
     }
 
+    /**
+     * Creates a rotation quaternion that rotates vector vecFrom to vector vecTo
+     * @param vecFrom defines the direction vector from which to rotate
+     * @param vecTo defines the direction vector to which to rotate
+     * @returns the new quaternion
+     */
+    public static FromUnitVectorsToRef(vecFrom: DeepImmutable<Vector3>, vecTo: DeepImmutable<Vector3>, result: Quaternion): Quaternion {
+        const r = Vector3.Dot(vecFrom, vecTo) + 1;
+
+        if (r < Epsilon) {
+            if (Math.abs(vecFrom.x) > Math.abs(vecFrom.z)) {
+                result.set(-vecFrom.y, vecFrom.x, 0, 0);
+            } else {
+                result.set(0, - vecFrom.z, vecFrom.y, 0);
+            }
+        } else {
+            const crossProduct = Vector3.Cross(vecFrom, vecTo);

```

Comment:
Something like Vector3.CrossToRef(vecFrom, vecTo, result)

?

Reply:
I must be missing something.
This would (probably) work in js, but typescript complains when passing quaternion as vector.

Reply:
I must be missing something.
This would (probably) work in js, but typescript complains about passing a quaternion as vector3

Reply:
I can maybe inline the cross product implementation? They do that in the three js as well.
Or I can cast the quaternion to any and pass it like that, but that doesn't feel right.

Reply:
this is what I meant:
```
 Vector3.CrossToRef(vecFrom, vecTo, result);
```

Reply:
oh wait.. I misread
so something like:
```
 Vector3.CrossToRef(vecFrom, vecTo, MathTmp.Vector3[0]);
```

Reply:
Nice, did not notice `MathTmp`.
I'll change it in a few hours.